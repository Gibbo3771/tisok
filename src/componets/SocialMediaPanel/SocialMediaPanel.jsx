// @flow
import React from "react";
import { FacebookShareButton, TwitterShareButton } from "react-share";
import { FacebookIcon, TwitterIcon } from "react-share";
import ReactTooltip from "react-tooltip";

export type Props = {
  url: string
};

export type State = {
  text: string,
  hashtags: Array<string>
};

export default class SocialMediaPanel extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = {
      text: "Advice to live by...",
      hashtags: ["tisok", "lifeprotips"]
    };
  }

  render() {
    const { url } = this.props;
    const { text, hashtags } = this.state;
    return (
      <div className="social-media-panel">
        <FacebookShareButton
          className="share-facebook"
          url={url}
          quote={text}
          hashtag={hashtags[0]}
        >
          <i
            class="fab fa-facebook-square icon-facebook icon"
            data-for="facebook-tooltip"
            data-tip="Share to Facebook"
          />
          <ReactTooltip
            className="tooltip"
            id="facebook-tooltip"
            place="top"
            effect="solid"
            type="tooltip"
          />
        </FacebookShareButton>
        <TwitterShareButton
          className="share-twitter"
          url={url}
          title={text}
          hashtags={hashtags}
        >
          <i
            class="fab fa-twitter-square icon-twitter icon"
            data-for="twitter-tooltip"
            data-tip="Share to Twitter"
          />
          <ReactTooltip
            className="tooltip"
            id="twitter-tooltip"
            place="top"
            effect="solid"
            type="tooltip"
          />
        </TwitterShareButton>
      </div>
    );
  }
}
