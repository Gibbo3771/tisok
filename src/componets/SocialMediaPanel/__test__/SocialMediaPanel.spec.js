import React from "react";
import SocialMediaPanel from "../SocialMediaPanel";

const props = {
  url: "www.google.com"
};

describe("SocialMediaPanel", () => {
  it("renders without crashing", () => {
    const wrapper = mount(<SocialMediaPanel url={props.url} />);
  });
  it("has can be found by id", () => {
    const wrapper = shallow(<SocialMediaPanel url={props.url} />);
    expect(wrapper.exists("#social-media-panel")).toBeDefined();
  });
  xit("should have a facebook button and icon", () => {
    const wrapper = shallow(<SocialMediaPanel url={props.url} />);
    expect(wrapper.exists(".share-facebook")).toBeTruthy();
    expect(wrapper.exists(".icon-facebook")).toBeTruthy();
  });
  it("should have a twitter button and icon", () => {
    const wrapper = shallow(<SocialMediaPanel url={props.url} />);
    expect(wrapper.exists(".share-twitter")).toBeTruthy();
    expect(wrapper.exists(".icon-twitter")).toBeTruthy();
  });
  it("should have some text to send to the social media platforms", () => {
    const wrapper = shallow(<SocialMediaPanel url={props.url} />);
    expect(wrapper.state().text).toBeDefined();
  });
  it("should have some hashtags to send to the social media platforms", () => {
    const wrapper = mount(<SocialMediaPanel url={props.url} />);
    expect(wrapper.state().hashtags).toBeDefined();
    expect(wrapper.state().hashtags.length).toBeGreaterThan(0);
  });
  it("should have some description to send to the social media platforms", () => {
    const wrapper = mount(<SocialMediaPanel url={props.url} />);
    expect(wrapper.state().description).toBeDefined();
  });
});
