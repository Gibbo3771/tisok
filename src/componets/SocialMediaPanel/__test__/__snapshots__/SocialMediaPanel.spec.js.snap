// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMediaPanel renders correctly 1`] = `
<SocialMediaPanel
  url="www.google.com"
>
  <div
    className="social-media-panel"
    data-for="social-media-tooltip"
    data-tip="Share"
  >
    <CreatedButton
      className="share-twitter"
      hashtags={
        Array [
          "tisok",
          "lifeprotips",
        ]
      }
      title="Advice to live by..."
      url="www.google.com"
      windowHeight={400}
      windowWidth={550}
    >
      <ShareButton
        className="share-twitter"
        disabledStyle={
          Object {
            "opacity": 0.6,
          }
        }
        hashtags={
          Array [
            "tisok",
            "lifeprotips",
          ]
        }
        network="twitter"
        networkLink={[Function]}
        openWindow={true}
        opts={
          Object {
            "hashtags": Array [
              "tisok",
              "lifeprotips",
            ],
            "title": "Advice to live by...",
            "via": undefined,
          }
        }
        role="button"
        tabIndex="0"
        title="Advice to live by..."
        url="www.google.com"
        windowHeight={400}
        windowWidth={550}
      >
        <div
          className="SocialMediaShareButton SocialMediaShareButton--twitter share-twitter"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          style={Object {}}
          tabIndex="0"
        >
          <i
            className="fab fa-twitter-square icon-twitter icon"
          />
        </div>
      </ShareButton>
    </CreatedButton>
    <ReactTooltip
      className="tooltip"
      clickable={false}
      effect="solid"
      id="social-media-tooltip"
      insecure={true}
      place="top"
      resizeHide={true}
      type="tooltip"
      wrapper="div"
    >
      <div
        className="__react_component_tooltip place-top type-dark "
        data-id="tooltip"
        id="social-media-tooltip"
      />
    </ReactTooltip>
  </div>
</SocialMediaPanel>
`;
